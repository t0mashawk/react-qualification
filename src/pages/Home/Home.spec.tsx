import React from "react";
import renderer from "react-test-renderer";

import HomeComponent, {filterRepos} from "./Home.component"

test("Homepage renders as expected", () => {
  const component = renderer.create(
    <HomeComponent/>
  );

  const tree = component.toJSON();
  
  expect(tree).toMatchSnapshot();
});


test('filterRepos() will reduce the repo array based on passed search term', () => {
  const dummyRepos = [
    {
      url: 'http://gogole.com',
      name: 'This is a name'
    },
    {
      url: 'http://gogole.com',
      name: 'another name'
    },
    {
      url: 'http://gogole.com',
      name: 'This is a name'
    },
    {
      url: 'http://gogole.com',
      name: 'another name'
    }
  ]
 
  const expectedResult: any = [
    {
      url: 'http://gogole.com',
      name: 'This is a name'
    },
    {
      url: 'http://gogole.com',
      name: 'This is a name'
    }
  ]

  const actualResult = filterRepos(dummyRepos, 'is')

  expect(actualResult).toEqual(expectedResult)
})