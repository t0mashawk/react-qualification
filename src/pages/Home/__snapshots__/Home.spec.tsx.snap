// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders as expected 1`] = `
<div
  className="sc-AxhCb szNEq"
>
  <div
    className="sc-AxhUy jjWtGP"
  >
    <input
      onChange={[Function]}
      value=""
    />
  </div>
  <p>
    Loading repos...
  </p>
  <div
    className="sc-AxirZ cmLbkl"
    color="navy"
  >
    <button
      onClick={[Function]}
    >
      Test
    </button>
  </div>
</div>
`;
